src/jdk.compiler/share/classes/com/sun/tools/javac/comp/ReflectMethods.java

```diff
@@ -901,6 +902,11 @@ public void visitTypeIdent(JCTree.JCPrimitiveTypeTree tree) {
             result = null;
         }
 
+        @Override
+        public void visitTypeArray(JCTree.JCArrayTypeTree tree) {
+            result = null; // MyType[].class is handled in visitSelect just as MyType.class
```

Comment:
Looks good - can you add a test case in `test/langtools/tools/javac/reflect/ConstantTest` (the last test case in this test has an example of class literals). Just to make sure the IR comes out ok, and that we don't have regression when we change the compiler code ;-)

Reply:
Good suggestion, done.